server/src/main/java/io/crate/execution/engine/aggregation/impl/ArbitraryAggregation.java

```diff
@@ -198,21 +203,26 @@ public Object partialResult(RamAccounting ramAccounting, MutableObject state) {
     private static class ArbitraryNumericDocValueAggregator extends SortedNumericDocValueAggregator<MutableObject> {
 
         private final DataType<?> columnDataType;
+        private final SizeEstimator<Object> sizeEstimator;
 
         public ArbitraryNumericDocValueAggregator(
             String columnName,
             DataType<?> columnDataType,
+            SizeEstimator<Object> sizeEstimator,
             CheckedBiConsumer<SortedNumericDocValues, MutableObject, IOException> docValuesConsumer
         ) {
-            super(columnName, MutableObject::new, docValuesConsumer);
+            super(columnName, (ramAccounting) -> new MutableObject(), docValuesConsumer);
             this.columnDataType = columnDataType;
+            this.sizeEstimator = sizeEstimator;
         }
 
         @Nullable
         @Override
         public Object partialResult(RamAccounting ramAccounting, MutableObject state) {
             if (state.hasValue()) {
-                return columnDataType.sanitizeValue(state.value());
+                var partialResult = columnDataType.sanitizeValue(state.value());
+                ramAccounting.addBytes(sizeEstimator.estimateSize(partialResult));
```

Comment:
Shouldn't we do this in `apply` instead, to account the used bytes earlier? Or does that make the abstractions more complex?

Reply:
Yes, forgot that it is called only once. I will just override the `apply` then and account for memory there.

Reply:
Hm it makes it a bit hard, it is not possible to reuse the abstraction, and extending it does not make sense. We need to have 4 different separate implementations for this case :/

Reply:
I'm a bit worried for the GROUP BY cases if we delay the accounting to `partialResult`. For aggregations it doesn't matter as it is only 1 row anyways. But afaik in the GROUP BY case the `partial` calls would happen at the very end, so we could have `unique_key_values * num_arbitrary_aggregation` aggregation states un-accounted for in-memory during the initial grouping phase, before any `arbitrary` state is accounted for.

Reply:
I see, let me look a bit more at it, but think for the arbitrary that's only the way to go with separate implementations or have smth like `ThreeConsumer` :)